### PR TITLE
refactor #1499: migrated notification usecase to kotlin

### LINKED ISSUE #1499: refactor: migrate notification package of core module to kotlin
migrate usecase/notification to kotlin

### PR DESCRIPTION
**Pull request by PratyushSingh07**
## Issue Fix
Fixes #1499 

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

File: core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/notification/FetchNotifications.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ */
+class FetchNotifications @Inject constructor(private val mFineractRepository: FineractRepository) :
+    UseCase<FetchNotifications.RequestValues?, FetchNotifications.ResponseValue?>() {
+
+    class RequestValues(val clientId: Long) : UseCase.RequestValues
+    class ResponseValue(
+        val notificationPayloadList: List<NotificationPayload?>
+    ) : UseCase.ResponseValue
+
+    override fun executeUseCase(requestValues: RequestValues?) {
```
Comment (therajanmaurya) [inline core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/notification/FetchNotifications.kt]: Remove `?`

File: core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/notification/FetchNotifications.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import rx.Subscriber
+import rx.android.schedulers.AndroidSchedulers
+import rx.schedulers.Schedulers
+import javax.inject.Inject
+
+/**
+ * Created by ankur on 24/July/2018
+ */
+class FetchNotifications @Inject constructor(private val mFineractRepository: FineractRepository) :
+    UseCase<FetchNotifications.RequestValues?, FetchNotifications.ResponseValue?>() {
```
Comment (therajanmaurya) [inline core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/notification/FetchNotifications.kt]: Remove this nullables `?`

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

File: core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/notification/FetchNotifications.kt (hunk 1 above)
Comment (PratyushSingh07) [reply core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/notification/FetchNotifications.kt]: done

File: core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/notification/FetchNotifications.kt (hunk 2 above)
Comment (PratyushSingh07) [reply core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/notification/FetchNotifications.kt]: done

**Comment by PratyushSingh07** [comment]
commited requested changes @therajanmaurya

**Pull request merged**